### PR TITLE
Add `hasDefaultValue` type trait

### PR DESCRIPTION
**Pull request by AmjadHD**
Needed for #21842.

### CODE REVIEW

File: compiler/semmagic.nim [hunk 1]
```diff
@@ -197,6 +197,8 @@ proc evalTypeTrait(c: PContext; traitCall: PNode, operand: PType, context: PSym)
     let complexObj = containsGarbageCollectedRef(t) or
                      hasDestructor(t)
     result = newIntNodeT(toInt128(ord(not complexObj)), traitCall, c.idgen, c.graph)
+  of "hasDefault":
+    result = newIntNodeT(toInt128(ord(not operand.requiresInit)), traitCall, c.idgen, c.graph)
```
Comment (ringabout) [inline compiler/semmagic.nim]: Could it be implemented as `hasRequiresInit` or something? A typetrait `hasDefault` might be used for types with default field names.

For example

```nim
type
  Foo = object
     id: int = 12
doAssert hasDefault(Foo)
```

Comment (AmjadHD) [reply compiler/semmagic.nim:201]: `hasDefault(Foo) == true` as it should be, isn't it ?

Comment (ringabout) [reply compiler/semmagic.nim]: But It might be expected

```nim
type
  Foo = object
     id: int
doAssert not hasDefault(Foo)
```

Comment (AmjadHD) [reply compiler/semmagic.nim]: We could call such function `hasDefaultField`.

**Comment by Araq** [comment]
Maybe `hasDefaultValue` is a better name.

File: compiler/semmagic.nim (hunk 1 above)
Comment (ringabout) [reply compiler/semmagic.nim]: Okay

**Pull request merged**